src/user-sessions/user-sessions.c

```diff
@@ -29,14 +29,9 @@ static int run(int argc, char *argv[]) {
         if (r < 0)
                 return r;
 
-        if (streq(argv[1], "start")) {
-                r = unlink_or_warn("/run/nologin");
-                k = unlink_or_warn("/etc/nologin");
-                if (r < 0)
-                        return r;
-                return k;
-
-        } else if (streq(argv[1], "stop"))
+        if (streq(argv[1], "start"))
+                return unlink_or_warn("/run/nologin");
```

Comment:
Can you add a comment here, explaining this:

```
/* Traditionally, there was only /etc/nologin, and we managed that. Then, in PAM $SOMEVERSION support for /run/nologin was added as alternative. 10y later we stopped managing /etc/nologin, leaving it for the administrator to manage. */

Also, please add README with the minimal PAM version we require now. Event if it's a super old one.

Reply:
I added a comment to create_shutdown_run_nologin_or_warn(), since that's where we create the file. And README already said PAM >= 1.1.2, which covers this. I amended the commit message.